lib/client.js

```diff
@@ -1199,6 +1203,334 @@ const PreClient = crypto => {
+        type,
+        data,
+        plainMeta,
+        fileMeta,
+        recipientSigningKey,
+        encryptionKeyPair,
+        {}, // Options not provided for anonymous notes
+        apiUrl
+      )
+    }
+
+    /**
+     * DeleteNote is a static method that deletes a note from TozStore based on the note identifier,
+     * allowing you to supply your own signingKey pair.
+     *
+     * @param {string} noteId  UUID assigned by TozStore, used to identify a note.
+     */
+    static async deleteNote(noteId, signingKeyPair, apiUrl = DEFAULT_API_URL) {
+      let anonAuth = await AuthRequest.anonymousAuth(
+        crypto,
+        signingKeyPair.publicKey,
+        signingKeyPair.privateKey,
+        apiUrl
+      )
+      return Client.internalDeleteNote(anonAuth, noteId, apiUrl)
+    }
+
+    /*
+     * Decrypts and validates a note response from TozStore given the proper keys.
+     */
+    static async decryptNoteJson(noteJson, publicSigningKey, privateKey, publicKey) {
+      const encryptedNote = Note.decode(noteJson)
+      const eak = noteJson.encrypted_access_key
+      const ak = await crypto.decryptNoteEak(privateKey, { eak: eak }, publicKey)
+      const decryptedNote = await crypto.decryptNote(encryptedNote, ak)
+
+      let signed = new SignedDocument(decryptedNote.core, decryptedNote.signature)
+      let verify = await Client.verify(signed, publicSigningKey)
+      if (!verify) {
+        throw new Error('Note failed verification')
+      }
+
+      return decryptedNote
+    }
+
+    /*
+     * InternalReadNote is an internal method used by internalReadNote
+     */
+    static async internalReadNote(authenticator, encryptionKeyPair, noteId, apiUrl) {
+      let request = await authenticator.tsv1Fetch(
+        apiUrl + `/v2/storage/notes?note_id=${noteId}`,
+        {
+          method: 'GET'
+        }
+      )
+      let storedNote = await checkStatus(request)
+      const noteJson = await storedNote.json()
+      return Client.decryptNoteJson(
+        noteJson,
+        authenticator.publicSigningKey,
+        encryptionKeyPair.privateKey,
+        encryptionKeyPair.publicKey
+      )
+    }
+
+    /*
+     * InternalReadNoteByName is an internal method used by readNoteByName
+     */
+    static async internalReadNoteByName(
+      authenticator,
+      encryptionKeyPair,
+      noteName,
+      apiUrl
+    ) {
+      let request = await authenticator.tsv1Fetch(
+        apiUrl + `/v2/storage/notes?id_string=${noteName}`,
+        {
+          method: 'GET'
+        }
+      )
+      let storedNote = await checkStatus(request)
+      const noteJson = await storedNote.json()
+      return Client.decryptNoteJson(
+        noteJson,
+        authenticator.publicSigningKey,
+        encryptionKeyPair.privateKey,
+        encryptionKeyPair.publicKey
+      )
+    }
+
+    /*
+     * InternalWriteNote is an internal method used by writeNote
+     */
+    static async internalWriteNote(
+      authenticator,
+      type,
```

Comment:
See comment above about some data items in notes that are less important than records. Some should potentially be in the options object, but if nothing else, reorder the params to put the least important last.